src/libponyrt/actor/actor.c

```diff
@@ -212,8 +212,20 @@ bool ponyint_actor_run(pony_ctx_t* ctx, pony_actor_t* actor, size_t batch)
       app++;
       try_gc(ctx, actor);
 
+      // if we become muted as a result of handling a message, bail out now.
+      if(actor->muted > 0)
```

Comment:
This should be an `atomic_load_explicit` with `memory_order_relaxed`. Atomic operations without an explicit memory order implicitly use `memory_order_seq_cst` and that's a performance hit.